### ISSUE TITLE
lambda error?

### ISSUE THREAD

**Issue by parmegv** [jorgenschaefer/elpy#606]
Error Message

u'repo_name'

Configuration

Virtualenv........: None
RPC Python........: 2.7.10 (/usr/bin/python)
Interactive Python: python (/usr/bin/python)
Emacs.............: 24.5.50.2
Elpy..............: Not found (Python), 1.8.0 (Emacs Lisp)
Jedi..............: 0.9.0
Rope..............: Not found (0.10.2 available)
Importmagic.......: Not found (0.1.3 available)
Syntax checker....: flake8 (/usr/bin/flake8)

Traceback

Traceback (most recent call last):
  File "/home/rafa/.emacs.d/elpa/elpy-20150502.740/elpy/jedibackend.py", line 303, in run_with_debug
  File "/usr/lib/python2.7/dist-packages/jedi/api/__init__.py", line 515, in call_signatures
    stmt = self._get_under_cursor_stmt(call_txt, start_pos)
  File "/usr/lib/python2.7/dist-packages/jedi/evaluate/cache.py", line 41, in wrapper
    rv = function(obj, *args, **kwargs)
  File "/usr/lib/python2.7/dist-packages/jedi/api/__init__.py", line 309, in _get_under_cursor_stmt
    user_stmt = self._parser.user_stmt()
  File "/usr/lib/python2.7/dist-packages/jedi/cache.py", line 143, in wrapper
    result = func(self)
  File "/usr/lib/python2.7/dist-packages/jedi/parser/user_context.py", line 293, in user_stmt
    module = self.module()
  File "/usr/lib/python2.7/dist-packages/jedi/parser/user_context.py", line 339, in module
    return self._parser().module
  File "/usr/lib/python2.7/dist-packages/jedi/cache.py", line 143, in wrapper
    result = func(self)
  File "/usr/lib/python2.7/dist-packages/jedi/parser/user_context.py", line 283, in _parser
    parser = FastParser(self._grammar, self._source, self._path)
  File "/usr/lib/python2.7/dist-packages/jedi/parser/fast.py", line 109, in __call__
    p.update(source)
  File "/usr/lib/python2.7/dist-packages/jedi/parser/fast.py", line 257, in update
    self._parse(source)
  File "/usr/lib/python2.7/dist-packages/jedi/parser/fast.py", line 383, in _parse
    line_offset, nodes)
  File "/usr/lib/python2.7/dist-packages/jedi/parser/fast.py", line 435, in _get_node
    p = Parser(self._grammar, parser_code, self.module_path, tokenizer=tokenizer)
  File "/usr/lib/python2.7/dist-packages/jedi/parser/__init__.py", line 155, in __init__
    self.module = p.parse(self._tokenize(tokenizer))
  File "/usr/lib/python2.7/dist-packages/jedi/parser/pgen2/parse.py", line 106, in parse
    if self.addtoken(type, value, prefix, start_pos):
  File "/usr/lib/python2.7/dist-packages/jedi/parser/pgen2/parse.py", line 170, in addtoken
    value, start_pos, prefix, self.addtoken)
  File "/usr/lib/python2.7/dist-packages/jedi/parser/__init__.py", line 269, in error_recovery
    self._stack_removal(grammar, stack, index + 1, value, start_pos)
  File "/usr/lib/python2.7/dist-packages/jedi/parser/__init__.py", line 304, in _stack_removal
    clear_names(children=node[1])
  File "/usr/lib/python2.7/dist-packages/jedi/parser/__init__.py", line 294, in clear_names
    clear_names(c.children)
  File "/usr/lib/python2.7/dist-packages/jedi/parser/__init__.py", line 298, in clear_names
    self._scope_names_stack[-1][c.value].remove(c)
KeyError: u'repo_name'

Jedi Debug Information

Jedi did not emit any debug info.

```

Reproduction:

```Python
import jedi

source = '''\
# -*- python -*-
# ex: set syntax=python:

# This is the dictionary that the buildmaster pays attention to. We also use
# a shorter alias to save typing.
c = BuildmasterConfig = {}

####### BUILDSLAVES

# We store the passwords for the buildslaves in a separate file, so we
# can share this one more widely. Thanks to https://svn.torproject.org/svn/projects/buildbot/trunk/master.cfg
PASSWORD_FILE = "passwords.py"

d = {}
execfile(PASSWORD_FILE, d)
PASSWORDS = d['PASSWORDS']
del d

# The 'slaves' list defines the set of recognized buildslaves. Each element is
# a BuildSlave object, specifying a unique slave name and password.  The same
# slave name and password must be configured on the slave.
from buildbot.buildslave import BuildSlave
slave_name = "localhost_slave"
c['slaves'] = [BuildSlave(slave_name, PASSWORDS[slave_name])]

# 'protocols' contains information about protocols which master will use for
# communicating with slaves.
# You must define at least 'port' option that slaves could connect to your master
# with this protocol.
# 'port' must match the value configured into the buildslaves (with their
# --master option)
c['protocols'] = {'pb': {'port': 9989}}

####### CHANGESOURCES

github_repos_username = 'parmegv'
github_repos_beginning = 'https://github.com/' + github_repos_username + '/'
default_branch = 'develop'
order_repos_index = 4
REPOS=[
    ('leap_pycommon', default_branch, github_repos_beginning + 'leap_pycommon.git', 1),
    ('keymanager', default_branch, github_repos_beginning + 'keymanager.git', 2),
    ('soledad', default_branch, github_repos_beginning + 'soledad.git', 3),
    ('leap_mail', default_branch, github_repos_beginning + 'leap_mail.git', 4)
]

# the 'change_source' setting tells the buildmaster how it should find out
# about source code changes.  Here we point to the buildbot clone of pyflakes.



from buildbot.changes.gitpoller import GitPoller
c['change_source'] = []
for repo_name, repo_branch, repo_url in REPOS:
    c['change_source'].append(GitPoller(
        repo_url,
        workdir='gitpoller-workdir-%s' % repo_name,
        branch=repo_branch,
        pollinterval=300))

####### SCHEDULERS

# Configure the Schedulers, which decide how to react to incoming changes.  In this
# case, just kick off a 'runtests' build

from buildbot.schedulers.basic import SingleBranchScheduler
from buildbot.schedulers.forcesched import ForceScheduler
from buildbot.changes import filter
c['schedulers'] = []

for repo_name, repo_branch, _ in REPOS:
    c['schedulers'].append(SingleBranchScheduler(
        name=repo_name,
        change_filter=filter.ChangeFilter(category=repo_name, branch=repo_branch),
        treeStableTimer=None,
        builderNames=[repo_name]))

c['schedulers'].append(ForceScheduler(
    name="force",
    builderNames=[repo[0] for repo in REPOS]))

####### BUILDERS

# The 'builders' list defines the Builders, which tell Buildbot how to perform a build:
# what steps, and which slaves can execute them.  Note that any particular build will
# only take place on one slave.

from buildbot.process.factory import BuildFactory
from buildbot.steps.source.git import Git
from buildbot.steps.shell import ShellCommand
from buildbot.config import BuilderConfig
from buildbot import locks

exclusive_slave_lock = locks.SlaveLock("exclusive")
c['builders'] = []

for repo_name, repo_branch, repo_url in sorted(REPOS, key = lambda repo[order_repos_index]: ## Origin of the error
    factory = BuildFactory()
    factory.addStep(Git(repourl=repo_url, mode='incremental', method='copy', branch=repo_branch, haltOnFailure=True))
    factory.addStep(ShellCommand(command=["python", "setup.py", "install"], haltOnFailure=True))
    factory.addStep(ShellCommand(command=["trial", "leap.mail"]))
    factory.addStep(ShellCommand(command=["trial", "leap.keymanager"]))        

    c['builders'].append(
	BuilderConfig(name=repo_name,
                      slavenames=[slave_name],
		      factory=factory, locks=[exclusive_slave_lock.access('exclusive')]))

####### STATUS TARGETS

# 'status' is a list of Status Targets. The results of each build will be
# pushed to these targets. buildbot/status/*.py has a variety to choose from,
# including web pages, email senders, and IRC bots.

c['status'] = []

from buildbot.status import html
from buildbot.status.web import authz, auth

webinterface_username = "web_interface"
authz_cfg=authz.Authz(
    # change any of these to True to enable; see the manual for more
    # options
    auth=auth.BasicAuth([(webinterface_username, PASSWORDS[webinterface_username])]),
    gracefulShutdown = False,
    forceBuild = 'auth', # use this to test your slave once it is set up
    forceAllBuilds = False,
    pingBuilder = False,
    stopBuild = False,
    stopAllBuilds = False,
    cancelPendingBuild = False,
)
c['status'].append(html.WebStatus(http_port=8010, authz=authz_cfg))

####### PROJECT IDENTITY

# the 'title' string will appear at the top of this buildbot
# installation's html.WebStatus home page (linked to the
# 'titleURL') and is embedded in the title of the waterfall HTML page.

c['title'] = "Bitmask Client"
c['titleURL'] = "https://github.com/leapcode/bitmask_client"

# the 'buildbotURL' string should point to the location where the buildbot's
# internal web server (usually the html.WebStatus page) is visible. This
# typically uses the port number set in the Waterfall 'status' entry, but
# with an externally-visible host name which the buildbot cannot figure out
# without some help.

c['buildbotURL'] = "http://localhost:8010/"

####### DB URL

c['db'] = {
    # This specifies what database buildbot uses to store its state.  You can leave
    # this at its default for all but the largest installations.
    'db_url' : "sqlite:///state.sqlite",
}
'''

script = jedi.Script(column=90, source=source, line=97, encoding='utf-8', path=u'/.../buildbot/master.cfg')
script.call_signatures()
```

**Comment by jorgenschaefer** [comment]
Thank you for the report!

Could you upgrade to Elpy 1.8.1 (released yesterday) and see if this problem persists?

**Comment by jorgenschaefer** [comment]
Lacking further updates, I assume this was fixed with Elpy 1.8.1. Thanks for the report, and please do not hesitate to open a new one if you have further problems!
